app/services/samples/metadata/update_service.rb

```diff
@@ -28,6 +28,8 @@ def execute
 
         @sample.save
 
+        update_metadata_summary
```

Comment:
So after going through the code I think we have a small issue. Right now the way it stands this will run the update_metadata_summary regardless of if a metadata field was `added` or `deleted`. This will also run if only values were `updated` which I don't think we want.

A possible solution is add a flag to the update_status hash which will be set to `true` if an addition or deletion action happened. Then in the `update_metadata_summary` we can return right away if the flag is false.

Reply:
Changes in 3ecf7d76649ad401bdaf20be265bd179130197ce with following notes:
- `update_status` now contains a key/value flags for `metadata_was_added` and `metadata_was_deleted`, with default value `false` and will update to `true` if those actions happen.
- `update_metadata_summary` will only occur if either of those are true (ie: skip if metadata was only `updated`)
- tests updated accordingly

Reply:
Logic was changed in 7d1d6d693cca394dbacd24cdd02a4e1a8593c0e7, see comment below for additional context.

We now check if any `metadata_fields` were `added` or `deleted`